### ISSUE TITLE
New failing lit tests / Rebase to the upcoming LLVM 14

### ISSUE THREAD

**Issue by michalpaszkowski** [KhronosGroup/LLVM-SPIRV-Backend#14]
The repository's default branch was switched to [feature/spirv-backend-llvm14](https://github.com/KhronosGroup/LLVM-SPIRV-Backend) which has all the SPIRV backend related changes rebased on top of the [3e2bd82f "Revert "[OptTable] Improve error message output for grouped short options"" commit from the LLVM trunk](https://github.com/llvm/llvm-project/commit/3e2bd82f).

Most of the problems coming from the rebase were solved in the commits referenced below:

1. [db04b750 "Fixes after rebase: Fixes to the commit 68befe3"](https://github.com/KhronosGroup/LLVM-SPIRV-Backend/commit/db04b7501af232667642f7368a05502909ea5662)
2. [c4557ff2 "Fixes after rebase"](https://github.com/KhronosGroup/LLVM-SPIRV-Backend/commit/c4557ff249f8d1aad3c2c6543ef8dd7394b15993)
3. [f7bdd8bf "Fixes after rebase: Fixes to commit d811090"](https://github.com/KhronosGroup/LLVM-SPIRV-Backend/commit/f7bdd8bfec60eb7b3cc467451bfdc94924e7bd12)
4. [5495374d "Fixes after rebase: Fixes to the commit d35e091"](https://github.com/KhronosGroup/LLVM-SPIRV-Backend/commit/5495374d62cd014bab332c30328e9f54861c866d)
5. [1732c070 "Fixes after rebase: Fixes to the commit 7565b9b"](https://github.com/KhronosGroup/LLVM-SPIRV-Backend/commit/1732c070829a7b4d94fe72f3a6d7e5da5cd3967d)
6. [d796e6c7 "Fixes after rebase: Fixes to the commit 5469c98"](https://github.com/KhronosGroup/LLVM-SPIRV-Backend/commit/d796e6c7b4eefdf20ab7fc99214fc57fd8e6582a)
7.  [d796e6c7 "Fixes after rebase: Fixes to the commit 5469c98"](https://github.com/KhronosGroup/LLVM-SPIRV-Backend/commit/d796e6c7b4eefdf20ab7fc99214fc57fd8e6582a)

Additionally, the following lit tests are failing due to the rebase and are still unresolved:

> LLVM :: CodeGen/SPIRV/instructions/atomic.ll
> LLVM :: CodeGen/SPIRV/instructions/atomic_acqrel.ll
> LLVM :: CodeGen/SPIRV/instructions/atomic_seq.ll
> LLVM :: CodeGen/SPIRV/instructions/fcmp.ll
> LLVM :: CodeGen/SPIRV/instructions/float-casts.ll
> LLVM :: CodeGen/SPIRV/instructions/icmp.ll
> LLVM :: CodeGen/SPIRV/instructions/integer-casts.ll
> LLVM :: CodeGen/SPIRV/instructions/intrinsics.ll
> LLVM :: CodeGen/SPIRV/instructions/ptrcmp.ll
> LLVM :: CodeGen/SPIRV/instructions/select.ll

**Comment by iliya-diyachkov** [comment]
All these tests fail in the same way:
```
#5 0x00005599152cbe2b llvm::RegisterClassInfo::runOnMachineFunction(llvm::MachineFunction const&) llvm/lib/CodeGen/RegisterClassInfo.cpp:63:36
#6 0x00005599151a269f (anonymous namespace)::MachineSinking::runOnMachineFunction(llvm::MachineFunction&) llvm/lib/CodeGen/MachineSink.cpp:432:8
#7 0x000055991508c3db llvm::MachineFunctionPass::runOnFunction(llvm::Function&) llvm/lib/CodeGen/MachineFunctionPass.cpp:72:33
#8 0x000055991574cdd4 llvm::FPPassManager::runOnFunction(llvm::Function&) llvm/lib/IR/LegacyPassManager.cpp:1439:20
#9 0x000055991574d09d llvm::FPPassManager::runOnModule(llvm::Module&) llvm/lib/IR/LegacyPassManager.cpp:1485:13
```
If I compile a test with -O0, it passes.

**Comment by iliya-diyachkov** [comment]
Also there are other new compile time fails on 5 tests:
1. EnqueueEmptyKernel.ll
```
...
 #7 0x0000557380d3d5d9 llvm::install_bad_alloc_error_handler(void (*)(void*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool), void*) llvm/lib/Support/ErrorHandling.cpp:130:61
 #8 0x000055737f4ad824 llvm::SPIRVTypeRegistry::generateOpenCLOpaqueType(llvm::StructType const*, llvm::MachineIRBuilder&, AccessQualifier::AccessQualifier) llvm/lib/Target/SPIRV/SPIRVTypeRegistry.cpp:493:1
 #9 0x000055737f4ac111 llvm::SPIRVTypeRegistry::handleOpenCLBuiltin(llvm::StructType const*, llvm::MachineIRBuilder&, AccessQualifier::AccessQualifier) llvm/lib/Target/SPIRV/SPIRVTypeRegistry.cpp:242:40
#10 0x000055737f4ac9b4 llvm::SPIRVTypeRegistry::createSPIRVType(llvm::Type const*, llvm::MachineIRBuilder&, AccessQualifier::AccessQualifier) llvm/lib/Target/SPIRV/SPIRVTypeRegistry.cpp:317:35
...
```
2. transcoding/BuildNDRange.ll, transcoding/BuildNDRange_2.ll, transcoding/annotate_attribute.ll, transcoding/fclamp.ll
```
...
 #7 0x0000564016c9c5d9 llvm::install_bad_alloc_error_handler(void (*)(void*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool), void*) llvm/lib/Support/ErrorHandling.cpp:130:61
 #8 0x00005640154bb5dd llvm::generateOpenCLBuiltinCall(llvm::StringRef, llvm::MachineIRBuilder&, llvm::Register, llvm::Type const*, llvm::SmallVectorImpl<llvm::Register> const&, llvm::SPIRVTypeRegistry*) llvm/lib/Target/SPIRV/SPIRVOpenCLBIFs.cpp:1130:1
 #9 0x0000564015488be2 llvm::SPIRVCallLowering::lowerCall(llvm::MachineIRBuilder&, llvm::CallLowering::CallLoweringInfo&) const llvm/lib/Target/SPIRV/SPIRVCallLowering.cpp:165:39
#10 0x000056401719d11f llvm::CallLowering::lowerCall(llvm::MachineIRBuilder&, llvm::CallBase const&, llvm::ArrayRef<llvm::Register>, llvm::ArrayRef<llvm::ArrayRef<llvm::Register> >, llvm::Register, std::function<unsigned int ()>) const llvm/lib/CodeGen/GlobalISel/CallLowering.cpp:149:19
#11 0x00005640171b598a llvm::IRTranslator::translateCallBase(llvm::CallBase const&, llvm::MachineIRBuilder&) llvm/lib/CodeGen/GlobalISel/IRTranslator.cpp:2306:21
...
```

**Comment by iliya-diyachkov** [comment]
These 8 tests are passed after 98890a9cc9f2b6e27d6caa25c144a0e9a4e930ff:
> LLVM :: CodeGen/SPIRV/instructions/atomic.ll
> LLVM :: CodeGen/SPIRV/instructions/atomic_acqrel.ll
> LLVM :: CodeGen/SPIRV/instructions/atomic_seq.ll
> LLVM :: CodeGen/SPIRV/instructions/float-casts.ll
> LLVM :: CodeGen/SPIRV/instructions/integer-casts.ll
> LLVM :: CodeGen/SPIRV/instructions/intrinsics.ll
> LLVM :: CodeGen/SPIRV/instructions/ptrcmp.ll
> LLVM :: CodeGen/SPIRV/instructions/select.ll

Other 2 tests fails with this problem:
> LLVM :: CodeGen/SPIRV/instructions/fcmp.ll
> LLVM :: CodeGen/SPIRV/instructions/icmp.ll
```
 #5 0x000055f333d726a4 llvm::Type::getTypeID() const llvm/include/llvm/IR/Type.h:135:37
 #6 0x000055f333d8af62 llvm::Type::isVectorTy() const llvm/include/llvm/IR/Type.h:238:46
 #7 0x000055f3349232b2 llvm::EVT::isExtendedVector() const llvm/lib/CodeGen/ValueTypes.cpp:77:1
 #8 0x000055f3349236dc llvm::EVT::isExtendedScalableVector() const llvm/lib/CodeGen/ValueTypes.cpp:116:29
 #9 0x000055f333d75cec llvm::EVT::isScalableVector() const llvm/include/llvm/CodeGen/ValueTypes.h:162:74
#10 0x000055f333d75dcb llvm::EVT::bitsLT(llvm::EVT) const llvm/include/llvm/CodeGen/ValueTypes.h:275:7
#11 0x000055f3348da14e llvm::TargetLoweringBase::getVectorTypeBreakdown(llvm::LLVMContext&, llvm::EVT, llvm::EVT&, unsigned int&, llvm::MVT&) const llvm/lib/CodeGen/TargetLoweringBase.cpp:1610:3
```

**Comment by iliya-diyachkov** [comment]
Fails on fcmp.ll and icmp.ll were fixed in 775051adefefbd737edf98c6a7e408ac83691df3.

**Comment by iliya-diyachkov** [comment]
After this 06888ab778838a926c4bdbd5a2384eb6a8901e22 commit EnqueueEmptyKernel.ll fails on missed OpenCL builtin function get_default_queue() (which is #6 ), so all extra compile fails are resolved after the rebase.
